### ISSUE TITLE
code2

### ISSUE THREAD

**Issue by Ruthwik333** [HarshitKmr10/somethingsomething#2]
*Testing Notebook*
**important**
For each function comment the function what is the function intend to do
comments hyper parameter tuning

** If i can setup a testing model just predict model wrapping the code into functions**

import pandas as pd
import numpy as np
import string
import re
import unicodedata
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.metrics import accuracy_score as acc
import lightgbm as lgb
import joblib

LE = LabelEncoder()
df = pd.read_excel("testdf.xlsx")
df['Meeting Notes'] = df['Meeting Notes'].astype(str)
df = df[['Meeting Notes', "Topic"]]
unique_topics = df['Topic'].unique()

def encoding(text):
    if isinstance(text, str):
        text = unicodedata.normalize("NFKD", text)
    return text

def remove_URL(text):
    return re.sub(r"https?://\S+|www.\S+", "", text)

def remove_non_ascii(text):
    return re.sub(r'[^\x00-\x7f]', r'', text)

def remove_html(text):
    html = re.compile(r"<.*?>|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});")
    return re.sub(html, "", text)

def remove_punct(text):
    return text.translate(str.maketrans('', '', string.punctuation))

df['Clean'] = df['Meeting Notes'].apply(remove_URL)
df['Clean'] = df['Clean'].apply(remove_non_ascii)
df['Clean'] = df['Clean'].apply(remove_html)
df["Topic"] = LE.fit_transform(df["Topic"])
x = df["Clean"]
y = df["Topic"]

lgb_model = joblib.load("model.pkl")

threshold = 0.75 

topic_names = LE.inverse_transform(np.arange(len(lgb_model.classes_)))
probabilities = lgb_model.predict_proba(x)
result_df = pd.DataFrame(probabilities, columns=topic_names)

predicted_classes = [topic_names[i] for i in np.argmax(probabilities, axis=1)]
filtered_classes = [cls if prob > threshold else 'NAN' for cls, prob in zip(predicted_classes, np.max(probabilities, axis=1))]

df["predictedtopic"] = filtered_classes
df2 = df.copy()
df2 = df2[df2["predictedtopic"] != "NAN"]
num_records_df2 = len(df2)  # Get the number of records in df2

# Print the number of records in df2
print("Number of records in df2:", num_records_df2)
train_acc = acc(LE.transform(df2["predictedtopic"]), df2['Topic'])
print("Accuracy: ", train_acc)

df = pd.concat([df, result_df], axis=1)
df["Topic"] = LE.inverse_transform(df["Topic"])
filtered_prob = [prob if prob > threshold else "NAN" for prob in np.max(probabilities, axis=1)]

df["probability"] = filtered_prob
# Inverse transform the encoded labels back to their original values

# Export the final DataFrame to an Excel file
df.to_excel('final_results.xlsx', index=False)
